src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -50,23 +39,154 @@ static ColliderCylinderInit D_80B18910 = {
     },
     { 20, 64, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B18360.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B183F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/EnTg_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/EnTg_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B185C0.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 0, 0, 0, 0, 0xFF };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/EnTg_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B18704.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B1871C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/func_80B18778.s")
+const ActorInit En_Tg_InitVars = {
+    ACTOR_EN_TG,
+    ACTORCAT_NPC,
+    FLAGS,
+    OBJECT_MU,
+    sizeof(EnTg),
+    (ActorFunc)EnTg_Init,
+    (ActorFunc)EnTg_Destroy,
+    (ActorFunc)EnTg_Update,
+    (ActorFunc)EnTg_Draw,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tg/EnTg_Draw.s")
+u16 func_80B18360(GlobalContext* globalCtx, Actor* thisx) {
+    EnTg* this = THIS;
+    u16 temp;
+    u32 phi;
+
+    temp = Text_GetFaceReaction(globalCtx, 0x24);
+    if (temp != 0) {
+        return temp;
+    }
+    if (globalCtx->sceneNum == 0x52) {
+        if (this->unk_208 & 1) {
+            phi = 0x5089;
+        } else {
+            phi = 0x508A;
+        }
+        return phi;
+    } else {
+        if (this->unk_208 & 1) {
+            phi = 0x7025;
+        } else {
+            phi = 0x7026;
+        }
+        return phi;
```

Comment:
I feel like this may have been covered before, but can these be ternaries?

Also,
- This is "really" doing `% 2 != 0` . Does that match?
- the purpose of `unk_208` is to serve different dialogue each successive time you talk to them (picking the first one randomishly based on frame). So something like `textSelect` maybe?

Reply:
* Yes, that does match.
* I named it `timesSpokenTo`.

Reply:
I think that's probably okay, although it doesn't quite capture it starting off as 0 or 1.